google-cloud-spanner-hibernate-dialect/src/main/java/com/google/cloud/spanner/hibernate/reflection/FieldKey.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2019-2020 Google LLC
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA
+ */
+
+package com.google.cloud.spanner.hibernate.reflection;
+
+import java.util.Objects;
+
+public class FieldKey {
+  private final Class<?> declaringClass;
+  private final Class<?> type;
+  private final String name;
+
+  public FieldKey(Class<?> type, String name) {
```

Comment:
I actually did this so we deliberately use less information than the `Field` object. This is because we would want  `@Id String id` to be treated as equal, even when declared on different classes (the parent vs the child) when doing the `childIds.containsAll(parentId)` check during validation

Reply:
Ahh got it, makes sense.